### ISSUE TITLE
Problem with new version 3.8.10

### ISSUE THREAD

**Issue by adit-s** [bubuntux/nordvpn#148]
### Before submitting review [Troubleshooting](https://github.com/bubuntux/nordvpn/wiki/Troubleshooting) and open/closed [Issues](https://github.com/bubuntux/nordvpn/issues?q=) ###
Reviewed

**Describe the bug**
With the same parameters, v1.0 works fine, but upgrading to 3.8.10 doesn't work. On start, it gives the error
"
Your account has expired. Renew your subscription now to continue enjoying the ultimate privacy and security with NordVPN.
You are not connected to NordVPN.
2021/02/12 17:04:42 releasing /etc/resolv.conf
"
NOTE: It works fine by excluding "--network=local-net-24".

**To Reproduce using docker CLI**
As follows with the only difference being using tag v1.0 or 3.8.10
"
docker run -ti --cap-add=NET_ADMIN --cap-add=SYS_MODULE --device /dev/net/tun --name vpn \
            --sysctl net.ipv4.conf.all.rp_filter=2 \
            --network=local-net-24 --restart=always \
            -e USER=xxxx@xxxx -e PASS='xxxxxxx' \
            -e CONNECT='United_States Seattle' -e TECHNOLOGY=NordLynx -d bubuntux/nordvpn
"

**To Reproduce without docker CLI**
N/A

**Expected behavior**
Expect the new version, 3.8.10, to start and connect a v1.0 does.

**Logs**
Logs from Portainer:
"
-A FORWARD -s 10.5.9.0/24 -j ACCEPT
,-A OUTPUT -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
,-A OUTPUT -o lo -j ACCEPT
,-A OUTPUT -o tap+ -j ACCEPT
,-A OUTPUT -o tun+ -j ACCEPT
,-A OUTPUT -o nordlynx -j ACCEPT
,-A OUTPUT -p udp -m udp --dport 53 -j ACCEPT
,-A OUTPUT -p udp -m udp --dport 51820 -j ACCEPT
,-A OUTPUT -p tcp -m tcp --dport 1194 -j ACCEPT
,-A OUTPUT -p udp -m udp --dport 1194 -j ACCEPT
,-A OUTPUT -p tcp -m tcp --dport 443 -j ACCEPT
,-A OUTPUT -d 10.5.9.0/24 -j ACCEPT
,
,nat:
,-P PREROUTING ACCEPT
,-P INPUT ACCEPT
,-P OUTPUT ACCEPT
,-P POSTROUTING ACCEPT
,-N DOCKER_OUTPUT
,-N DOCKER_POSTROUTING
,-A POSTROUTING -o tap+ -j MASQUERADE
,-A POSTROUTING -o tun+ -j MASQUERADE
,-A POSTROUTING -o nordlynx -j MASQUERADE
,
,mangle:
,-P PREROUTING ACCEPT
,-P INPUT ACCEPT
,-P FORWARD ACCEPT
,-P OUTPUT ACCEPT
,-P POSTROUTING ACCEPT
,
,raw:
,-P PREROUTING ACCEPT
,-P OUTPUT ACCEPT
,
,
,2021/02/12 17:04:42 Get "https://w569ut7zbkiqf5b.xyz/v1/users/services": dial tcp: lookup w569ut7zbkiqf5b.xyz on 127.0.0.11:53: read udp 127.0.0.1:54688->127.0.0.11:53: read: connection refused
,2021/02/12 17:04:42 [Error] listing services: ntp check failed
,
  
Opening the web browser...
,If nothing happens, please visit https://join.nordvpn.com/order/?utm_medium=app&utm_source=linux
,
-2021/02/12 17:04:42 POST_CONNECT system info:
,Routing Table:
,default via 10.5.9.1 dev eth0 
,10.5.9.0/24 dev eth0 proto kernel scope link src 10.5.9.24 
,
,IP Rules:
,0:	from all lookup local 
,32766:	from all lookup main 
,32767:	from all lookup default 
,
,IP Tables:
,filter:
,-P INPUT DROP
,-P FORWARD DROP
,-P OUTPUT DROP
,-A INPUT -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
,-A INPUT -i lo -j ACCEPT
,-A INPUT -s 10.5.9.0/24 -j ACCEPT
,-A FORWARD -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
,-A FORWARD -i lo -j ACCEPT
,-A FORWARD -d 10.5.9.0/24 -j ACCEPT
,-A FORWARD -s 10.5.9.0/24 -j ACCEPT
,-A OUTPUT -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
,-A OUTPUT -o lo -j ACCEPT
,-A OUTPUT -o tap+ -j ACCEPT
,-A OUTPUT -o tun+ -j ACCEPT
,-A OUTPUT -o nordlynx -j ACCEPT
,-A OUTPUT -p udp -m udp --dport 53 -j ACCEPT
,-A OUTPUT -p udp -m udp --dport 51820 -j ACCEPT
,-A OUTPUT -p tcp -m tcp --dport 1194 -j ACCEPT
,-A OUTPUT -p udp -m udp --dport 1194 -j ACCEPT
,-A OUTPUT -p tcp -m tcp --dport 443 -j ACCEPT
,-A OUTPUT -d 10.5.9.0/24 -j ACCEPT
,
,nat:
,-P PREROUTING ACCEPT
,-P INPUT ACCEPT
,-P OUTPUT ACCEPT
,-P POSTROUTING ACCEPT
,-N DOCKER_OUTPUT
,-N DOCKER_POSTROUTING
,-A POSTROUTING -o tap+ -j MASQUERADE
,-A POSTROUTING -o tun+ -j MASQUERADE
,-A POSTROUTING -o nordlynx -j MASQUERADE
,
,mangle:
,-P PREROUTING ACCEPT
,-P INPUT ACCEPT
,-P FORWARD ACCEPT
,-P OUTPUT ACCEPT
,-P POSTROUTING ACCEPT
,
,raw:
,-P PREROUTING ACCEPT
,-P OUTPUT ACCEPT
,
,
,
  
Your account has expired. Renew your subscription now to continue enjoying the ultimate privacy and security with NordVPN.
,
-
  

-
  
You are not connected to NordVPN.
,

2021/02/12 17:04:42 releasing /etc/resolv.conf
,
"

**Additional context**
Same results on Ubuntu 18.04 and VMware Photon 3.0 (both upto date).
Ubuntu running docker version 18.06.1-ce
VMware Photon running docker version 19.03.10

**Comment by adit-s** [comment]
local-net-24 is a macvlan network to "connect" the container to the "local" subnet (using the .24 address)

docker network inspect local-net-24
[
    {
        "Name": "local-net-24",
        "Id": "82095cb5273b325d00ebf7aaceada12512545f1a12ac5d1fe36814b4a1e33679",
        "Created": "2021-02-12T12:24:12.910121637-05:00",
        "Scope": "local",
        "Driver": "macvlan",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": [
                {
                    "Subnet": "10.5.9.0/24",
                    "IPRange": "10.5.9.24/32",
                    "Gateway": "10.5.9.1"
                }
            ]
        },
        "Internal": false,
        "Attachable": true,
        "Ingress": false,
        "ConfigFrom": {
            "Network": "local-net-24-base"
        },
        "ConfigOnly": false,
        "Containers": {
            "9abe546b4d261ca4cbed025a27f385323d94c5d96a532c8b16f9e02111207a18": {
                "Name": "seattle-vpn",
                "EndpointID": "3f4724b94bc4272979400d6dd3b549a31033b1d2f625ee543f96069637806906",
                "MacAddress": "02:42:0a:05:09:18",
                "IPv4Address": "10.5.9.24/24",
                "IPv6Address": ""
            }
        },
        "Options": {
            "parent": "eth0"
        },
        "Labels": {}
    }
]

**Comment by clp71s** [comment]
This issue may be caused by the start_vpn.sh script flushing the nat iptable rules Docker uses for internal DNS name resolution to 127.0.0.11. These rules are not present when using the default bridge network in Docker, therefore the issue does not exhibit under those conditions.

I found that commenting out line 8 of /usr/bin/start_vpn.sh (iptables -t nat -F) resulted in a successful connection to NordVPN. However, I'm not sure if removing this step of the killswitch may result in DNS leakage. I also found that manually setting my nameserver record in /etc/resolv.conf to the bridge network gateway IP address solved the issue. Perhaps the script can be altered to take either option into account?

The following rules are present in the nat iptable when I put my container on its own network bridge:

-P PREROUTING ACCEPT
-P INPUT ACCEPT
-P OUTPUT ACCEPT
-P POSTROUTING ACCEPT
-N DOCKER_OUTPUT
-N DOCKER_POSTROUTING
-A OUTPUT -d 127.0.0.11/32 -j DOCKER_OUTPUT
-A POSTROUTING -d 127.0.0.11/32 -j DOCKER_POSTROUTING
-A DOCKER_OUTPUT -d 127.0.0.11/32 -p tcp -m tcp --dport 53 -j DNAT [unsupported revision]
-A DOCKER_OUTPUT -d 127.0.0.11/32 -p udp -m udp --dport 53 -j DNAT [unsupported revision]
-A DOCKER_POSTROUTING -s 127.0.0.11/32 -p tcp -m tcp --sport 40929 -j SNAT --to-source :53
-A DOCKER_POSTROUTING -s 127.0.0.11/32 -p udp -m udp --sport 40330 -j SNAT --to-source :53
